etcdserver/cluster.go

```diff
@@ -59,6 +59,11 @@ type Cluster struct {
 	id    types.ID
 	token string
 	store store.Store
+	// index is the raft index that cluster is updated at bootstrap
+	// from remote cluster info. It avoids snapshot recovery overwriting the
+	// transport setting, then blocking the communication.
+	// TODO: upgrade it as last modified index
+	index uint64
```

Comment:
we should also mention that the cluster might be from a future view in the view of local etcd server